src/state_machine.zig

```diff
@@ -594,6 +745,653 @@ test "linked accounts" {
+        .input = std.mem.zeroInit(Commit, .{
+            .id = 1,
+            .timestamp = timestamp_cont,
+            .flags = .{ .padding = 1 },
+        }),
+    }, CommitTestVector{
+        .outcome = CommitTransferResult.transfer_not_found,
+        .input = std.mem.zeroInit(Commit, .{
+            .id = 777,
+            .timestamp = timestamp_cont,
+        }),
+    }, CommitTestVector{
+        .outcome = CommitTransferResult.transfer_not_two_phase_commit,
+        .input = std.mem.zeroInit(
+            Commit,
+            .{ .id = 1, .timestamp = timestamp_cont },
+        ),
+    }, CommitTestVector{
+        .outcome = CommitTransferResult.ok,
+        .input = std.mem.zeroInit(
+            Commit,
+            .{ .id = 2, .timestamp = timestamp_cont },
+        ),
+    }, CommitTestVector{
+        .outcome = CommitTransferResult.already_committed_but_accepted,
+        .input = std.mem.zeroInit(
+            Commit,
+            .{
+                .id = 2,
+                .timestamp = (timestamp_cont + 1),
+                .flags = .{ .reject = true },
+            },
+        ),
+    }, CommitTestVector{
+        .outcome = CommitTransferResult.already_committed,
+        .input = std.mem.zeroInit(
+            Commit,
+            .{
+                .id = 2,
+                .timestamp = (timestamp_cont + 1),
+            },
+        ),
+    }, CommitTestVector{
+        .outcome = CommitTransferResult.ok,
+        .input = std.mem.zeroInit(Commit, .{
+            .id = 3,
+            .timestamp = (timestamp_cont + 1),
+            .flags = .{ .reject = true },
+        }),
+    }, CommitTestVector{
+        .outcome = CommitTransferResult.already_committed_but_rejected,
+        .input = std.mem.zeroInit(Commit, .{
+            .id = 3,
+            .timestamp = (timestamp_cont + 2),
+        }),
+    }, CommitTestVector{
+        .outcome = CommitTransferResult.transfer_expired,
+        .input = std.mem.zeroInit(Commit, .{
+            .id = 4,
+            .timestamp = (timestamp_cont + 2),
+        }),
+    }, CommitTestVector{
+        .outcome = CommitTransferResult.condition_requires_preimage,
+        .input = std.mem.zeroInit(Commit, .{
+            .id = 5,
+            .timestamp = (timestamp_cont + 2),
+        }),
+    }, CommitTestVector{
+        .outcome = CommitTransferResult.preimage_invalid,
+        .input = std.mem.zeroInit(Commit, .{
+            .id = 5,
+            .timestamp = (timestamp_cont + 2),
+            .reserved = [_]u8{1} ** 32,
+            .flags = .{
+                .preimage = true,
+            },
+        }),
+    }, CommitTestVector{
+        .outcome = CommitTransferResult.preimage_requires_condition,
+        .input = std.mem.zeroInit(Commit, .{
+            .id = 6,
+            .timestamp = (timestamp_cont + 2),
+            .flags = .{
+                .preimage = true,
+            },
+        }),
+    } };
+
+    // Test Balance BEFORE commit
```

Comment:
```suggestion
    // Test balances BEFORE commit:
```

In general, comments should be like sentences and either end with a colon if it refers to what follows or a full stop if it refers to something that's a complete standalone unit within a branch, or if it's a function comment.